ansible_collections/arista/avd/plugins/action/anta_catalog_builder.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+from typing import Any, Dict, List, Tuple
+
+from ansible.plugins.action import ActionBase
+from anta.loader import parse_catalog
+
+TestTuple = Tuple[str, Dict[str, str]]
+Catalog = Dict[str, List[Dict[str, Any]]]
+
+
+AVD_MAPPING: Dict[str, List[TestTuple]] = {
+    "management_api_http": [("anta.tests.security.VerifyAPIHttpStatus", {}), ("anta.tests.security.VerifyAPIHttpsSSL", {"profile": "https_ssl_profile"})]
+}
+
+
+def _get_package_name(test_name: str) -> str:
+    """Extracts the package name from a fully qualified test name."""
+    return test_name.rsplit(".", 1)[0]
+
+
+def _get_test_name(test_name: str) -> str:
+    """Extracts the test name from a fully qualified test name."""
+    return test_name.rsplit(".", 1)[1]
+
+
+def _update_test_parameters(value: Any, test_parameters: Dict[str, str]) -> Dict[str, Any]:
+    """Updates the test parameters based on the provided structured configuration values."""
+    return {param_key: value.get(structured_config_key) for param_key, structured_config_key in test_parameters.items()}
+
+
+def _build_new_catalog(task_vars: Dict[str, Any], test_mapping: Dict[str, List[TestTuple]]) -> Catalog:
+    """
+    Constructs package tests from the structured configuration and the test mapping.
+
+    Returns:
+        A dictionary containing package tests.
+    """
+    new_catalog: Catalog = {}
+    for key, value in task_vars.items():
```

Comment:
task vars can be _very_ large, so I would suggest to flip this to loop over test_mappings and check if the key is there.

Reply:
Thinking more about this, maybe it would be cleaner to build a small function/class for each of the ANTA tests which accepts the structured config as input, and extracts the stuff we need. I think it would be simpler to maintain over time.

Reply:
Yeah I had in mind to create a dynamic catalog depending on the task_vars. If we decide to extract the stuff we need in the ANTA tests directly, we will have to create the test catalog manually.